lib/instrumentation/kafkajs/producer.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const { MessageSpec } = require('../../shim/specs')
+
+module.exports = function instrumentProducer({ shim, kafkajs }) {
+  shim.wrap(kafkajs.Kafka.prototype, 'producer', function nrProducerWrapper(shim, orig) {
+    return function nrProducer() {
+      const params = shim.argsToArray.apply(shim, arguments)
+      const producer = orig.apply(this, params)
+
+      // The `.producer()` method returns an object with `send` and `sendBatch`
+      // methods. The `send` method is merely a wrapper around `sendBatch`, but
+      // we cannot simply wrap `sendBatch` because the `send` method does not
+      // use the object scoped instance (i.e. `this.sendBatch`); it utilizes
+      // the closure scoped instance of `sendBatch`. So we must wrap each
+      // method.
+
+      shim.recordProduce(producer, 'send', function nrSend(shim, fn, n, args) {
+        const data = args[0]
+        return new MessageSpec({
+          destinationName: data.topic,
+          destinationType: shim.TOPIC
```

Comment:
I think you'll have to include headers so we can injected our DT/CAT headers

Reply:
Headers is optional according to our messaging/kafka spec. I removed adding them because the headers are unique per message, e.g.:

```js
producer.send({
  topic: 'topic',
  messages: [
    { key: 'key', value: 'value', headers: {"stuff"},
    { key: 'key', value: 'value', headers: {"diff stuff"},
  ]
})
```

Reply:
that's fine if they'e optional but you need to pass them in so we can add our headers and I think we're going to have to build logic to insert per message headers otherwise we're not going to get DT